gokitmiddlewares/loggingmiddleware/middleware.go

```diff
@@ -0,0 +1,67 @@
+package loggingmiddleware
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/arquivei/foundationkit/errors"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/rs/zerolog/log"
+)
+
+// New returns a new go-kit logging middleware with the given name and configuration.
+// It will panic if the name is empty.
+func New(name string, c Config) endpoint.Middleware {
+	if name == "" {
+		panic("endpoint name is empty")
+	}
+
+	if c.Logger == nil {
+		panic("logger is nil")
+	}
+
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, req interface{}) (resp interface{}, err error) {
+			begin := time.Now()
+
+			ctx = initLoggerContext(ctx, *c.Logger)
+
+			enrichLoggerContext(ctx, name, c, req)
```

Comment:
Nit: `enrichLoggerContext`, `doLogging` and `enrichLoggerAfterResponse` all access the context to obtain the logger. Since the logger is so mingled with this middleware and these functions are unexported, they could receive the `logger` itself so that less context operations are executed.